angular-tour-of-heroes/src/app/shared/services/mailservice.ts

```diff
@@ -0,0 +1,30 @@
+import { HttpClient, HttpErrorResponse, HttpParams } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { catchError, Observable, tap, throwError } from 'rxjs';
+import { environment } from 'src/environments/environment';
+import { IGenre } from '../models/IGenre';
+import { ObserversModule } from '@angular/cdk/observers';
+import { IMailData } from '../models/IMailData';
+
+
+const apiUrl=environment.apiUrl;
+
+@Injectable({
+    providedIn:"root"
+})
+
+export class MailService{
+
+    constructor(
+        private http:HttpClient
+    ){}
+
+
+    sendEmail(mailData:IMailData):Observable<any>{
```

Comment:
You should only use "any" when you are unsure of what you will get and then replace it with an actual interface once you test.
However, there are cases where you simply must use any. For example: error handling can sometimes throw multiple types of errors, some of which are unexpected. You can use any there, especially if errors are only logged and not displayed to the user as such.